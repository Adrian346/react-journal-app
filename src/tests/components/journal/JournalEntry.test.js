import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';


const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}

let store = mockStore(initState)
store.dispatch = jest.fn()

const nota = {
    id: 1,
    date: 0,
    body: 'Hola',
    title: 'mundo',
    url: 'https://algun-lugar.com/foto.jpg'
}

const wrapper = mount(
    <Provider store={ store }>
        <JournalEntry { ...nota } />
    </Provider>
)


describe('Pruebas en JournalEntry', () => {

    test('Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe de activar la note', () => {
        wrapper.find('.journal__entry').prop('onClick')()
        expect(store.dispatch).toHaveBeenCalledWith(
            activeNote(nota.id, { ...nota } )
        )        
    })
    
    
})
