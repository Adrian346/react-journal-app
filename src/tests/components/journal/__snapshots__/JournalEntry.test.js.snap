// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en JournalEntry Debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate_faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algun-lugar.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      mundo
    </p>
    <p
      className="journal__entry-content"
    >
      Hola
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
       
      Wednesday
       
    </span>
    <h4>
       
      31st
       
    </h4>
  </div>
</div>
`;
